sympy/integrals/manualintegrate.py

```diff
@@ -831,16 +842,33 @@ def quadratic_denom_rule(integral):
 
 def sqrt_quadratic_denom_rule(integral: IntegralInfo):
     integrand, x = integral
-    numer, denom = integrand.as_numer_denom()
     a = Wild('a', exclude=[x])
     b = Wild('b', exclude=[x])
     c = Wild('c', exclude=[x, 0])
-    match = denom.match(sqrt(a+b*x+c*x**2))
+    numer = Wild('numer')
+    match = integrand.match(numer/sqrt(a+b*x+c*x**2))
     if match:
+        a, b, c, numer = match[a], match[b], match[c], match[numer]
         numer_poly = numer.as_poly(x)
         if numer_poly is None:
             return
-        a, b, c = match[a], match[b], match[c]
+        degenerate_cond = Eq(c, 0)
+        if degenerate_cond is S.false:
+            degenerate_step = None
+        else:
+            degenerate_integrand = numer/sqrt(a+b*x)
+            next_degenerate_cond = Eq(b, 0)
+            next_degenerate_step = None if next_degenerate_cond is S.false else integral_steps(numer/sqrt(a), x)
+            if next_degenerate_cond is S.true:
+                degenerate_step = next_degenerate_step
+            else:
+                u = Dummy("u")
+                substep = integral_steps(numer.subs(x, (u**2-a)/b)*2/b, u)
+                degenerate_step = URule(u, sqrt(a+b*x), None, substep, degenerate_integrand, x)
+                if next_degenerate_step:
+                    degenerate_step = PiecewiseRule([(next_degenerate_step, next_degenerate_cond),
+                                                     (degenerate_step, S.true)], degenerate_integrand, x)
+        denom = sqrt(a+b*x+c*x**2)
```

Comment:
This seems quite convoluted. I'm sure there's a simpler way to write this. I would have thought something like:
```python
if Eq(c, 0) is not S.false:
    degenerate_case = integral_steps(numer/sqrt(a+b*x), x)
    step = Piecewise((step, Ne(c, 0)), (degenerate_case, True))
```

Maybe this is why degenerate cases should be handled by recursion so we don't need to explicitly list degenerate cases inside degenerate cases.

Reply:
Recursively call `integral_steps` instead of a specific rule is time consuming. Here since we already know the structure of the integrand, we are more or less in the same situation like https://github.com/sympy/sympy/pull/23379#discussion_r852386054.
To make the code cleaner, it may worth have a rule to integrate `poly/sqrt(a+b*x)`. Currently it's too slow
```py
In [4]: %timeit integrate((x**3+1)/sqrt(x+1), manual=True)
4.46 s ± 30.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
and produces many unnecessary steps https://sympy-beta.vercel.app/input/integrate((x**3+1)%2Fsqrt(x+1))